es-core/src/components/TextComponent.cpp

```diff
@@ -73,32 +90,32 @@ void TextComponent::render(const Eigen::Affine3f& parentTrans)
 {
 	Eigen::Affine3f trans = parentTrans * getTransform();
 
-	/*Eigen::Vector3f dim(mSize.x(), mSize.y(), 0);
-	dim = trans * dim - trans.translation();
-	Renderer::pushClipRect(Eigen::Vector2i((int)trans.translation().x(), (int)trans.translation().y()), 
-		Eigen::Vector2i((int)(dim.x() + 0.5f), (int)(dim.y() + 0.5f)));
-		*/
+	if (mBgColor)
+	{
+		Renderer::drawRect(getPosition().x(), getPosition().y() - 1,
+			getSize().x(), getSize().y(), mBgColor);
+	}
 
-	if(mTextCache)
+	if (mTextCache)
 	{
 		const Eigen::Vector2f& textSize = mTextCache->metrics.size;
 		Eigen::Vector3f off(0, (getSize().y() - textSize.y()) / 2.0f, 0);
 
-		if(Settings::getInstance()->getBool("DebugText"))
+		if (Settings::getInstance()->getBool("DebugText"))
 		{
 			// draw the "textbox" area, what we are aligned within
 			Renderer::setMatrix(trans);
 			Renderer::drawRect(0.f, 0.f, mSize.x(), mSize.y(), 0xFF000033);
 		}
-		
+
 		trans.translate(off);
 		trans = roundMatrix(trans);
 		Renderer::setMatrix(trans);
 
 		// draw the text area, where the text actually is going
-		if(Settings::getInstance()->getBool("DebugText"))
+		if (Settings::getInstance()->getBool("DebugText"))
 		{
-			switch(mAlignment)
+			switch (mAlignment)
 			{
```

Comment:
There is a bunch of whitespace reformatting in this file that should be done separately (or not done depending on other file styling).